EXT_BeamPhysics.md

```diff
@@ -0,0 +1,346 @@
+```
+
+Note: The `SpeciesType` extension must be used when using the `BeamPhysics` extension.
+
+Definitions
+-----------
+
+- **Lattice**: A **lattice** is the arrangement of elements in a machine such as a particle accelerator.
+
+- **Lattice branches**: The lattices of some programs can contain multiple connected beam lines or rings. For example, an injection line connected to a storage ring connected to X-ray beam lines. Each line or ring is called a **branch**. In the above example, the injection line is one branch, the storage ring another branch, and each X-ray beam line is a branch.
+
+- **Global coordinate system**: The **global** coordinate system is a coordinate system that is used to describe the position and orientation of machine elements in space. That is, the **global** coordinate system is fixed with respect to the building or room where the machine is placed independent of the machine itself.
+
+- **Lattice coordinate system**: The curvilinear coordinate system whose "longitudinal" coordinate (**s**) typically runs through the nominal centers of the elements in the machine. The **lattice** coordinate system is often used to describe particle positions.
+
+- **Macro-particle**: Macro-particles are simulation particles that represent multiple real particles. The number of real particles that a macro-particle represents affects the calculation of the field due to a macro-particle but does not affect tracking.
+
+- **Particle coordinate system**: The coordinate system with respect to which particles positions and momenta are given. Some programs use the **global coordinate system** as the **particle coordinate system** while other programs use a coordinate system that is derived from the curvilinear **lattice coordinate system**.
+
+- **Polar coordinates**: **Polar** coordinates are **(r, theta, phi)** where **r** is the radius, **theta** is the angle from the **z** or **s** axis, and **phi** is the projected azimuthal angle in the **(x, y)** plane.
+
+- **Particle Group**: The **Particle Group** is a group for specifying a set of particles such as the particles in a bunch. The Beam Physics extension defines a standard for the  **Particle Group** as discussed below.
+
+Notes:
+------
+
+- When using the **lattice** coordinate system, the `position` coordinates are **(x, y, s)** or **(x, y, z)** where, nominally, **x** is the "horizontal" component, **y** is the "vertical" coordinate, and **s** or **z** is the lattice longitudinal coordinate.
+
+Additional File Root Group (Path `/`) Attributes
+------------------------------------------------
+
+The following records are defined for the file root group.
+
+- `fileType`
+  - Type: Optional *(string)*
+  - Description: The type of data being stored in the file. If present, must be set to `openPMD`. This attribute is used in systems where different data files can contain different types of data and allows for quick identification of the what type of data is in a given file.
```

Comment:
maybe I missed it, but isn't this already possible to be quickly identified with the base standard identifier, attribute `openPMD` in `/`?

Reply:
But a non-openPMD file will not have the `openPMD` attribute.

Reply:
I am confused how this is useful for files that use openPMD and the openPMD extension :)

Reply:
It is useful for readers that can read a variety of data types including openPMD.